data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/CsaProfileRemedialActionsCreator.java

```diff
@@ -159,10 +158,26 @@ private void addOnContingencyStateUsageRules(RemedialActionAdder<?> remedialActi
                 .withUsageMethod(usageMethod).add());
     }
 
-    private void checkAllContingenciesLinkedToRaHaveTheSameConstraintKind(String remedialActionId, Set<PropertyBag> linkedContingencyWithRAs, String firstKind) {
+    private void checkElementCombinationConstraintKindsCoherence(String remedialActionId, Set<PropertyBag> linkedContingencyWithRAs) {
+        // The same contingency cannot have different Element Combination Constraint Kinds
+        Map<String, Boolean> numberOfIncludedPerContingency = new HashMap<>();
```

Comment:
Actually, the same contingency can appear several times and. Besides, we cannot know for sure the order in which the contingency will appear. The reason I used maps instead of booleans is that we need to have a persistent data storage for each contingency since it may be met several times.

Bu the coherence check is done at each loop to break as soon as an error has been spotted.

Reply:
As you use these 2 maps only to add value "true", you can replace them by 2 sets, each one containing contingencies with included/considered. Also, you should rename them, as they don't contain the number of included/considered, by example numberOfIncludedPerContingency -> setContingenciesWithIncluded